roles/openshift_health_checker/openshift_checks/logging/logging_index_time.py

```diff
@@ -0,0 +1,175 @@
+        """Add log entry by making unique request to Kibana. Check for unique entry in the ElasticSearch pod logs."""
+        pod_log_timeout = int(get_var(task_vars, "openshift_check_pod_logs_timeout", default=ES_CMD_TIMEOUT_SECONDS))
+
+        # get all Kibana pods
+        self.logging_namespace = get_var(task_vars, "openshift_logging_namespace", default=self.logging_namespace)
+        pods, error = super(LoggingIndexTime, self).get_pods_for_component(
+            self.execute_module,
+            self.logging_namespace,
+            "kibana",
+            task_vars,
+        )
+
+        if error:
+            return {"failed": True, "changed": False, "msg": error}
+
+        running_kibana_pods = self.running_pods(pods)
+        if not len(running_kibana_pods):
+            msg = ('No Kibana pods were found to be in the "Running" state.'
+                   'At least one Kibana pod is required in order to perform this check.')
+            return {"failed": True, "changed": False, "msg": msg}
+
+        # get all Elasticsearch pods
+        pods, error = super(LoggingIndexTime, self).get_pods_for_component(
+            self.execute_module,
+            self.logging_namespace,
+            "es",
+            task_vars,
+        )
+
+        if error:
+            return {"failed": True, "changed": False, "msg": error}
+
+        running_es_pods = self.running_pods(pods)
+        if not len(running_es_pods):
+            msg = ('No Elasticsearch pods were found to be in the "Running" state.'
+                   'At least one Elasticsearch pod is required in order to perform this check.')
+            return {"failed": True, "changed": False, "msg": msg}
+
+        self.check_log_delay(running_kibana_pods[0], running_es_pods[0], pod_log_timeout, task_vars)
+        return {}
+
+    def check_log_delay(self, kibana_pod, es_pod, pod_log_timeout, task_vars):
+        """Check delay of logs and return any errors accordingly"""
+        uid = self.curl_kibana_with_uuid(kibana_pod, task_vars)
+        self.wait_until_cmd_or_err(es_pod, uid, pod_log_timeout, task_vars)
+
+    def wait_until_cmd_or_err(self, es_pod, uid, timeout_secs, task_vars):
+        """Wait a maximum of timeout_secs for the uuid logged in Kibana to be found in
+        the Elasticsearch logs. Since we are querying for a message with the unique
+        uuid from earlier, there should only be a single match.
+        Raise an OpenShiftCheckException if the timeout is reached before a result is found."""
+
+        interval = 1  # seconds to wait between retries
+        timeout_orig = timeout_secs
+
+        while timeout_secs > 0:
+            time.sleep(interval)
+            timeout_secs -= interval
+
+            time_start = int(time.time())
+            total = self.query_es_from_es(es_pod, uid, task_vars)
+            time_end = int(time.time())
+            if total == 1:
+                return
+
+            timeout_secs -= (time_end - time_start)
```

Comment:
Copying discussion from 29149c2a6d246a1dc91e6e751f9312be74c45cfd:

@rhcarvalho 
> This is buggy, because time.time() is not monotonic. time_end - time_start may be negative, the clock may have 
> changed between the two calls.
> See https://github.com/golang/proposal/blob/master/design/12914-monotonic.md#computer-clocks.
>
> We may decide to ignore the bug, but then we could as well ignore the time it takes to run query_es_from_es and 
> know that timeout_secs is not to be taken with precision.

@juanvallejo
> Thanks for the link to that proposal.
> What if we keep the time it takes to run query_es_from_es, but only count it against timeout_secs whenever 
> (time_end - time_start) > -1? That way, we keep a precise duration for timeout_secs 99...% of the time